src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -339,7 +339,7 @@ public Worker(String session, ServerBuilder<?> serverBuilder, ShardWorkerConfig
       throws ConfigurationException {
     super("BuildFarmShardWorker");
     this.config = config;
-    isCasShard = !config.getOmitFromCas();
+    isCasShard = !config.getCapabilities().getOmitFromCas();
```

Comment:
That will help clean up these logics to be a little less awkward.

Reply:
fixed